osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -275,27 +278,36 @@ protected override void LogoArriving(OsuLogo logo, bool resuming)
 
                 sideFlashes.Delay(FADE_IN_DURATION).FadeIn(64, Easing.InQuint);
             }
-            else if (!api.IsLoggedIn || api.State.Value == APIState.RequiresSecondFactorAuth)
+            else
             {
                 // copy out old action to avoid accidentally capturing logo.Action in closure, causing a self-reference loop.
                 var previousAction = logo.Action;
 
-                // we want to hook into logo.Action to display the login overlay, but also preserve the return value of the old action.
+                // we want to hook into logo.Action to display certain overlays, but also preserve the return value of the old action.
                 // therefore pass the old action to displayLogin, so that it can return that value.
                 // this ensures that the OsuLogo sample does not play when it is not desired.
-                logo.Action = () => displayLogin(previousAction);
+                logo.Action = () => onLogoClick(previousAction);
             }
+        }
 
-            bool displayLogin(Func<bool> originalAction)
+        private bool onLogoClick(Func<bool> originalAction)
+        {
+            if (!api.IsLoggedIn || api.State.Value == APIState.RequiresSecondFactorAuth)
             {
                 if (!loginDisplayed.Value)
                 {
-                    Scheduler.AddDelayed(() => login?.Show(), 500);
+                    this.Delay(500).Schedule(() => login?.Show());
```

Comment:
Why?

Reply:
Reads better in my eyes, delay 500 then do X. I guess you don't agree.

Reply:
I disagree with changing things for no reason. Plus, we use the previous method 99% of the time, so I'd rather continue to do that.

Reply:
Just to clarify on the "no reason" part, I've used the new method in the display of the new dialog, and since now there are two different ways of delaying an action in a single method, I chose to update the old existing line as well. I don't do things for no reason.

Reply:
nah